#! /usr/bin/env python

PACKAGE = 'bitbots_motion'
import roslib

roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
gen.add("dyn_falling_active", bool_t, 1,
        "False: Servos will be turned of when falling; true: robot tries to fall dynamically", False)
gen.add("ground_coefficient", double_t, 1,
        "Hardness of the floor, 1.25 for artificial turf(soft), 1 for hard ground", min=0, max=2)

gen.add("threshold_gyro_y_front", double_t, 1,
        "Threshold harder -> earlier reaction, but more false positives.", min=-3.14, max=3.14)
gen.add("threshold_gyro_y_back", double_t, 1,
        "Threshold harder -> earlier reaction, but more false positives.", min=-3.14, max=3.14)
gen.add("threshold_gyro_x_right", double_t, 1,
        "Threshold harder -> earlier reaction, but more false positives.", min=-3.14, max=3.14)
gen.add("threshold_gyro_x_left", double_t, 1,
        "Threshold harder -> earlier reaction, but more false positives.", min=-3.14, max=3.14)

exit(gen.generate(PACKAGE, "bitbots_motion", "motion_params"))
